### ISSUE TITLE
Error on Hardlink, navigation and extending document\page

### ISSUE THREAD

**Issue by Yivan** [pimcore/pimcore#655]
We just get an error related to DAO, in a specific use case on last Pimcore release (4.1.1). We extended document page (**using namespace "project"**, and using pimocreNavigation to generate a menu. If the menu is made with some links, all is ok.
But converting one of this link to hardlink, bring an error.
Keeping this hadlink and disabling classmap.php (so Document\Page is not more extended), there is no more error.
So this is a mix of hardlink/page extending (and error occurs when calling pimcoreNavigation).

As you can see it try to include Project\Document\Page\Dao.php.

```
#0  Project\handleError(2, include_once(Project\Document\Page\Dao.php): failed to open stream: No such file or directory, C:\ServeurLocal\domains\dev-project.local\www\vendor\zendframework\zendframework1\library\Zend\Loader.php, 134, ...
#1  Zend_Loader::loadFile() called at [C:\ServeurLocal\domains\dev-project.local\www\vendor\zendframework\zendframework1\library\Zend\Loader.php:134]
#2  Zend_Loader::loadFile(Project\Document\Page\Dao.php, , 1) called at [C:\ServeurLocal\domains\dev-project.local\www\vendor\zendframework\zendframework1\library\Zend\Loader.php:82]
#3  Zend_Loader::loadClass(Project\Document\Page\Dao)
#4  call_user_func(Array ([0] => Zend_Loader,[1] => loadClass), Project\Document\Page\Dao) called at [C:\ServeurLocal\domains\dev-project.local\www\vendor\zendframework\zendframework1\library\Zend\Loader\Autoloader.php:479]
#5  Zend_Loader_Autoloader->_autoload(Project\Document\Page\Dao)
#6  call_user_func(Array ([0] => Zend_Loader_Autoloader Object ...
#7  Zend_Loader_Autoloader::autoload(Project\Document\Page\Dao)
#8  spl_autoload_call(Project\Document\Page\Dao) called at [C:\ServeurLocal\domains\dev-project.local\www\pimcore\lib\Pimcore\Model\AbstractModel.php:139]
#9  Pimcore\Model\AbstractModel->initDao(Project\Document\Page) called at [C:\ServeurLocal\domains\dev-project.local\www\pimcore\models\Document\Hardlink\Service.php:45]
#10 Pimcore\Model\Document\Hardlink\Service::wrap(Project\Document\Page Object ...
#11 Pimcore\Model\Document\Hardlink\Wrapper\Link->getChilds() called at [C:\ServeurLocal\domains\dev-project.local\www\pimcore\lib\Pimcore\View\Helper\PimcoreNavigationController.php:203]
#12 Pimcore\View\Helper\PimcoreNavigationController->getChilds(Pimcore\Model\Document\Hardlink\Wrapper\Link Object ...
#13 Pimcore\View\Helper\PimcoreNavigationController->buildNextLevel(Pimcore\Model\Document\Hardlink\Wrapper\Link Object ...
#14 Pimcore\View\Helper\PimcoreNavigationController->buildNextLevel(Pimcore\Model\Document\Folder Object ...
#15 Pimcore\View\Helper\PimcoreNavigationController->getNavigation(Project\Document\Page Object ...
#16 Pimcore\View\Helper\PimcoreNavigation->pimcoreNavigation(Project\Document\Page Object ...
#17 call_user_func_array(Array ([0] => Pimcore\View\Helper\PimcoreNavigation Object ...
```

In my view (error is triggered by this line):

```
<?php
echo $this->pimcoreNavigation($this->document, $node, 'header-menu1-')->menu()->setUseTranslator(false)->renderMenu(null, [
    "maxDepth" => 1,
     "ulClass" => "menu-head"
]);
?>
```

In classmap.php

```
return [
    "Document\\Page" => "Project\\Document\\Page"
];
```

I don't think it is code related as it works fine with link. But as soon their is a hard link, the DAO seems to try to include a file of the extended Page class (wich doesn't exist). I think i sould look for pimcore class and not "project" class.
Maybe somewhere in the core code, a backslash is missing so it tries to resolve on the "project" namespace...

Feel free to ask any other information if needed. Thanks.

**Comment by brusch** [comment]
Should be fixed in build 3829 (available in a few minutes).

**Comment by brusch** [comment]
fix see also 501d44f63374e3d90e991a3c98c60920dd5472b0
